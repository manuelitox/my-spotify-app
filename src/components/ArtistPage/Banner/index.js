import React from 'react'

import StyledBanner from './styles'
import BannerWrapper from './wrapper'
import StylesWaveArtistIcon from './stylesWaiveIcon'

const Banner = () => (
  <StyledBanner>
    <BannerWrapper />
    <StylesWaveArtistIcon />
  </StyledBanner>
)

export default Banner