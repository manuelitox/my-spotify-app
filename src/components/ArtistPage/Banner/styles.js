import styled from 'styled-components'

const StyledBanner = styled.div`
  background-color: ${ props => props.theme.blueLight };
  margin-bottom: 80px;
  padding: 60px 0 100px;
  position: relative;
  @media only screen and (max-width: ${ props => props.theme.size.medium }) {
    padding: 40px 0;
  }         
  @media only screen and (max-width: ${ props => props.theme.size.small }) {
    padding: 30px 0;
  }     
  & > div {
    display: grid;
    grid-gap: 30px;
    grid-template-columns: 120px 1fr;
    @media only screen and (max-width: ${ props => props.theme.size.medium }) {
      grid-template-columns: 100px 1fr;
    }         
    @media only screen and (max-width: ${ props => props.theme.size.small }) {
      grid-template-columns: 1fr;
    }         
  }
  img {
    border-radius: 100%;
    @media only screen and (max-width: ${ props => props.theme.size.medium }) {
      height: 100px;
      width: 100px;
    }  
  }
  h1 {
    color: white;
    font-weight: 500;
    font-size: 6rem;
    margin: 0;
    @media only screen and (max-width: ${ props => props.theme.size.medium }) {
      font-size: 5rem;
    }     
    @media only screen and (max-width: ${ props => props.theme.size.small }) {
      font-size: 3rem;
    }     
  }
  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }
  li {
    display: inline-block;
    color: ${ props => props.theme.blueLightText };
    font-size: 2rem;
    font-weight: 500;
    text-transform: uppercase;
    @media only screen and (max-width: ${ props => props.theme.size.medium }) {
      font-size: 1.5rem;
    }      
    @media only screen and (max-width: ${ props => props.theme.size.small }) {
      font-size: 1rem;
    }         
  }
  span {
    margin: 0 10px;
    @media only screen and (max-width: ${ props => props.theme.size.small }) {
      margin: 0 4px;
    }     
  }
`

export default StyledBanner