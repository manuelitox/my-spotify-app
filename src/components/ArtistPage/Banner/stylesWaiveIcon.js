import styled from 'styled-components'

import { ReactComponent as WaveArtistIcon } from 'svgs/bgWaveArtist.svg'

const StylesWaveArtistIcon = styled(WaveArtistIcon)`
  height: 80px;
  position: absolute;
  top: 100%;
  width: 100%;
  @media only screen and (max-width: ${ props => props.theme.size.small }) {
    height: 40px;
  }    
`

export default StylesWaveArtistIcon