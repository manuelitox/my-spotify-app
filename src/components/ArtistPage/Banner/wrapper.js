import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import WithContainer from 'components/generic/Container/'

const BannerWrapper = ({
  name,
  photo,
  genres
}) => (
  <Fragment>
    <img alt={ name } src={ photo } height="120" width="120" />
    <div>
      <h1>{ name }</h1>
      <ul>
        { genres.map( (genre, index) => {
          return <li key={ index }>{ genre } { index + 1 < genres.length ? <span>-</span> : null }</li>
        }) }
      </ul>
    </div>
  </Fragment>
)

BannerWrapper.defaultProps = {
  name: 'Alicia Keys',
  photo: 'https://i.scdn.co/image/05e25d030a1caa1611b1e7248d89b557d9867258',
  genres: [
    'hip pop',
    'neo soul',
    'pop',
    'r&b',
    'urban contemporary'
  ]
}

BannerWrapper.propTypes = {
  name: PropTypes.string.isRequired,
  photo: PropTypes.string,
  genres: PropTypes.array.isRequired
}

export default WithContainer(BannerWrapper)