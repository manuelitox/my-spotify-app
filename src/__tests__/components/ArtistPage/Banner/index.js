import React from 'react'
import { shallow } from 'enzyme'

import Banner from 'components/ArtistPage/Banner/'

describe('Component | Banner', () => {
  let component 
  beforeEach( () => {
    component = shallow(
      <Banner />
    )
  })
  it('should render a component', () => {
    expect(component).toMatchSnapshot()
  })
})