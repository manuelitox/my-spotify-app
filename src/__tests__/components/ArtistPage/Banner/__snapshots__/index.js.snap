// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | Banner should render a component 1`] = `
<styled.div>
  <Component />
  <Styled(ReactComponent) />
</styled.div>
`;
